x/staking/keeper/historical_info.go

```diff
@@ -79,9 +99,9 @@ func (k Keeper) TrackHistoricalInfo(ctx sdk.Context) {
 	// Since the entries to be deleted are always in a continuous range, we can iterate
 	// over the historical entries starting from the most recent version to be pruned
 	// and then return at the first empty entry.
-	for i := ctx.BlockHeight() - int64(entryNum); i >= 0; i-- {
-		_, found := k.GetHistoricalInfo(ctx, i)
-		if found {
+	for i := sdkCtx.BlockHeight() - int64(entryNum); i >= 0; i-- {
+		_, err := k.GetHistoricalInfo(ctx, i)
+		if err == nil {
 			k.DeleteHistoricalInfo(ctx, i)
 		} else {
```

Comment:
I think we should return the error directly if `!errors.Is(err, types.ErrNoHistoricalInfo)` instead of breaking from the loop. The error is just discarded otherwise.